pkg/volume/provision.go

```diff
@@ -447,8 +475,24 @@ func (p *nfsProvisioner) getServer() (string, error) {
 		return "", fmt.Errorf("service %s=%s is valid but it doesn't have a cluster IP", p.serviceEnv, serviceName)
 	}
 
-	glog.Infof("using service %s=%s cluster IP %s as NFS server IP", p.serviceEnv, serviceName, service.Spec.ClusterIP)
-	return service.Spec.ClusterIP, nil
+	// do the nodes support "cluster.local" internal DNS names?
+	// TODO maybe we can auto-detect this, or make this user-configurable?
+	nodes_support_cluster_local := false
```

Comment:
The problem is that `cluster.local` domain can be overridden by any other value. So we can't simple autodetect this.